Update podman docs (#32252)
DataDog/documentation: content/en/containers/guide/podman-support-with-docker-integration.md

1. Run one of the following commands to deploy the Agent.
   1. To deploy the Agent without logging:

      **Note**: Replace `<CONTAINERS_PATH>` with the path to your containers directory, `<API_KEY>` with your API key, and `<DD_HOSTNAME>` with your Datadog hostname:
      ```shell
      podman run -d --name dd-agent \
        --cgroupns host --pid host \
        -v <CONTAINERS_PATH>:/var/lib/containers:ro \
        -v /proc/:/host/proc/:ro \
        -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
        -e DD_API_KEY=<API_KEY> \
        -e DD_HOSTNAME=<DD_HOSTNAME> \
        gcr.io/datadoghq/agent:latest
      ```

    1. To deploy the Agent with log collection run the agent as follows:
      ```shell
      podman run -d --name dd-agent \
          --cgroupns host --pid host \
          -v <CONTAINERS_PATH>:/var/lib/containers:ro \
          -v /proc/:/host/proc/:ro \
          -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
          -e DD_API_KEY=<API_KEY> \
          -e DD_HOSTNAME=<DD_HOSTNAME> \
          -e DD_LOGS_ENABLED=true \
          -e DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL=true \
          -e DD_LOGS_CONFIG_USE_PODMAN_LOGS=true \
          gcr.io/datadoghq/agent:latest

>       ```

Reviewer: ```suggestion
       ```shell
       podman run -d --name dd-agent \
           --cgroupns host --pid host \
           -v <CONTAINERS_PATH>:/var/lib/containers:ro \
           -v /proc/:/host/proc/:ro \
           -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
           -e DD_API_KEY=<API_KEY> \
           -e DD_HOSTNAME=<DD_HOSTNAME> \
           -e DD_LOGS_ENABLED=true \
           -e DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL=true \
           -e DD_LOGS_CONFIG_USE_PODMAN_LOGS=true \
           gcr.io/datadoghq/agent:latest
       ```
```

## Requirements

* Podman version >= 3.2.0
* Datadog Agent version >= 7.54.0

## Host Agent installation

Host Agent installation requires quite a few manual steps and permission tweaks for the Agent to be able to collect the data. Those fixes are not always permanent and are required again if containers restart.
We advice to follow `rootles` and `rootful` installation instead since it is easier to perform and maintain.

## Agent deployment as a Podman rootless container

To deploy the Agent as a rootless Podman container, the command to run is similar to the one used for [Docker][2].

The main difference is that as the Agent does not need to have access to the runtime socket, it relies on the Podman DB to extract the container information that it needs. Instead of mounting the Docker socket and setting `DOCKER_HOST`, the only thing needed is to mount the Podman DB location into the container.

Setting up Podman DB path depends on the Agent version.

> **Agent < v7.54.0**

Reviewer: ```suggestion
### Agent < v7.54.0
```

User needs to set the path to Podman DB (`<PODMAN_DB_PATH>` in the command below).

> To discover the exact location of Podman DB, run the following command:

Reviewer: ```suggestion
Find the location of Podman DB
1. Run the following command:
```

```
$ podman info -f json | jq '.store.graphRoot'
"$HOME/.local/share/containers/storage"

> ```

Reviewer: ~~~suggestion
   ```shell
   $ podman info -f json | jq '.store.graphRoot'
   "$HOME/.local/share/containers/storage"
```
~~~

> Check which DB you have at that path and set `<PODMAN_DB_PATH>` accordingly.

Reviewer: ~~~suggestion
1. Set the database at that path to the environment variable `<PODMAN_DB_PATH>`. Replace `path/to/db` with the path to the database:
   ```shell
   export PODMAN_DB_PATH=path/to/db
   ```
~~~
It's useful to give people the command to speed things up. Is a temporary mapping like this okay?

Before version 4.8 Podman uses BoltDB as the default database backend.
In some systems the path of the Podman DB is `$HOME/.local/share/containers/storage/libpod/bolt_state.db` but it might be different in your system. Set `<PODMAN_DB_PATH>` in the command below accordingly.

```
$ podman run -d --name dd-agent \
    --cgroupns host --pid host \
    -v <PODMAN_DB_PATH>:/var/lib/containers/storage/libpod/bolt_state.db:ro \
    -v /proc/:/host/proc/:ro \
    -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
    -e DD_API_KEY=<API_KEY> \
    -e DD_HOSTNAME=<DD_HOSTNAME> \
    gcr.io/datadoghq/agent:latest

> ```

Reviewer: ~~~suggestion
   ```
   $ podman run -d --name dd-agent \
       --cgroupns host --pid host \
       -v $PODMAN_DB_PATH:/var/lib/containers/storage/libpod/bolt_state.db:ro \
       -v /proc/:/host/proc/:ro \
       -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
       -e DD_API_KEY=<API_KEY> \
       -e DD_HOSTNAME=<DD_HOSTNAME> \
       gcr.io/datadoghq/agent:latest
   ```
~~~
They've set the environment variable above. Can they just do this?
